### PR TITLE
added-json-header

### PR DESCRIPTION
**Pull request by IvaTutis**
A small PR on the FASTA header writer to enable writing of FASTA's with the new JSON header as discussed for the new GFF3 pipelines

### CODE REVIEW

File: src/main/java/uk/ac/ebi/embl/fasta/writer/FastaFileWriter.java [hunk 1]
```diff
@@ -70,12 +71,45 @@ public void write() throws IOException {
...
       case TRANSLATION_HEADER_FORMAT:
         header = String.format(">%s", entry.getPrimaryAccession());
+      case JSON_FASTA_HEADER:
+        header =
+            String.format(
+                ">%s | { \"description\":\"%s\", \"molecule_type\":\"%s\", \"topology\":\"%s\"}",
+                getEffectiveAccession(entry), // same logic as in Gff3Tools
+                entry.getDescription().getText(),
+                entry.getSequence().getMoleculeType(),
+                entry.getSequence().getTopology());
```
Comment (Rajkumar-D) [inline src/main/java/uk/ac/ebi/embl/fasta/writer/FastaFileWriter.java]: I think you should add a null check for all these values. I suggest moving that to a method as well. 
```
entry.getDescription().getText(),
 entry.getSequence().getMoleculeType(),
 entry.getSequence().getTopology());
```

Also add test for the null cases

**Review by Rajkumar-D — COMMENTED** [review]
Added comment

File: src/main/java/uk/ac/ebi/embl/fasta/writer/FastaFileWriter.java [hunk 2]
```diff
@@ -70,12 +74,74 @@ public void write() throws IOException {
...
+    }
+    if (moleculeType != null) {
+      jsonFields.add(String.format("\"molecule_type\":\"%s\"", moleculeType));
+    }
+    if (topology != null) {
+      jsonFields.add(String.format("\"topology\":\"%s\"", topology));
+    }
+    String jsonPart = jsonFields.isEmpty() ? "{}" : "{ " + String.join(", ", jsonFields) + " }";
+    return jsonPart;
+  }
```
Comment (Rajkumar-D) [inline src/main/java/uk/ac/ebi/embl/fasta/writer/FastaFileWriter.java]: Add following dependency
`implementation("com.fasterxml.jackson.core:jackson-databind:2.18.6")
`

and update the method to: 
 ```
 private String getJsonFastaHeader(Entry entry) {
    Map<String, String> json = new LinkedHashMap<>();

    if (entry.getDescription() != null) {
      json.put("description", entry.getDescription().getText());
    }

    if (entry.getSequence() != null) {
      if (entry.getSequence().getMoleculeType() != null) {
        json.put("molecule_type", entry.getSequence().getMoleculeType());
      }
      if (entry.getSequence().getTopology() != null) {
        json.put("topology", entry.getSequence().getTopology().toString());
      }
    }

    try {
      String jsonPart = OBJECT_MAPPER.writeValueAsString(json);
      return String.format(">%s | %s", getEffectiveAccession(entry), jsonPart);
    } catch (JsonProcessingException e) {
      throw new IllegalStateException("Failed to build JSON part", e);
    }
  }
```

File: src/main/java/uk/ac/ebi/embl/fasta/writer/FastaFileWriter.java [hunk 3]
```diff
@@ -70,12 +74,74 @@ public void write() throws IOException {
...
         header = String.format(">%s", entry.getPrimaryAccession());
+      case JSON_FASTA_HEADER:
+        String jsonPart = getJsonPart();
+
+        // make header
+        header =
+            String.format(
+                ">%s | %s",
+                getEffectiveAccession(entry), // same logic as in Gff3Tools
+                jsonPart);
```
Comment (Rajkumar-D) [inline src/main/java/uk/ac/ebi/embl/fasta/writer/FastaFileWriter.java]: Suggestion to do this:
`  header = getJsonFastaHeader(entry);`

**Review by Rajkumar-D — APPROVED** [review]
